SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp

```diff
@@ -1,9 +1,859 @@
+	if (SchemaDatabase == nullptr)
+	{
+		bDatabaseIsValid = false;
+	}
+	else
+	{
+		for (const auto& CurrentClass : Classes)
+		{
+			if (!TestEqualDatabaseEntryAndSchemaFile(CurrentClass, gSchemaOutputFolder, SchemaDatabase))
+			{
+				bDatabaseIsValid = false;
+				break;
+			}
+		}
+	}
+
+	TestTrue("Generated schema database is valid", bDatabaseIsValid);
+
+	// CLEANUP
+	DeleteTestFolders();
+
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_schema_database_exists_WHEN_schema_database_deleted_THEN_no_schema_database_exists)
+{
+	// GIVEN
+	UClass* CurrentClass = ASpatialTypeActor::StaticClass();
+	TSet<UClass*> Classes = { CurrentClass };
+
+	SpatialGDKEditor::Schema::ResetSchemaGeneratorState();
+	SpatialGDKEditor::Schema::SpatialGDKGenerateSchemaForClasses(Classes, gSchemaOutputFolder);
+	SpatialGDKEditor::Schema::SaveSchemaDatabase(gDatabaseOutputFile);
+
+	// WHEN
+	SpatialGDKEditor::Schema::DeleteSchemaDatabase(gSchemaDatabaseFileName );
+
+	// THEN
+	const FString SchemaDatabasePackagePath = FPaths::Combine(FPaths::ProjectContentDir(), gSchemaDatabaseFileName);
+	const FString ExpectedgSchemaDatabaseFileName = FPaths::SetExtension(SchemaDatabasePackagePath, FPackageName::GetAssetPackageExtension());
+	IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+	TestFalse("Generated schema database does not exists", PlatformFile.FileExists(*ExpectedgSchemaDatabaseFileName));
+
+	// CLEANUP
+	DeleteTestFolders();
+
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_schema_database_exists_WHEN_tried_to_load_THEN_loaded)
+{
+	// GIVEN
+	UClass* CurrentClass = ASpatialTypeActor::StaticClass();
+	TSet<UClass*> Classes = { CurrentClass };
+
+	SpatialGDKEditor::Schema::ResetSchemaGeneratorState();
+	SpatialGDKEditor::Schema::SpatialGDKGenerateSchemaForClasses(Classes, gSchemaOutputFolder);
+	SpatialGDKEditor::Schema::SaveSchemaDatabase(gDatabaseOutputFile);
+
+	// WHEN
+	bool bSuccess = SpatialGDKEditor::Schema::TryLoadExistingSchemaDatabase(gSchemaDatabaseFileName);
+
+	// THEN
+	TestTrue("Schema database loaded", bSuccess);
+
+	// CLEANUP
+	DeleteTestFolders();
+
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_schema_database_does_not_exist_WHEN_tried_to_load_THEN_not_loaded)
+{
+	// GIVEN
+	AddExpectedError(TEXT("Attempted to delete schema database"), EAutomationExpectedErrorFlags::Contains, 1);
+	SpatialGDKEditor::Schema::DeleteSchemaDatabase(gSchemaDatabaseFileName );
+
+	// WHEN
+	const FString SchemaDatabasePackagePath = FPaths::Combine(FPaths::ProjectContentDir(), gSchemaDatabaseFileName);
+	const FString ExpectedgSchemaDatabaseFileName = FPaths::SetExtension(SchemaDatabasePackagePath, FPackageName::GetAssetPackageExtension());
+	// TODO(Alex): it deleted schema folder :(
+	//bool bSuccess = SpatialGDKEditor::Schema::TryLoadExistingSchemaDatabase(gSchemaDatabaseFileName);
+	bool bSuccess = false;
+
+	// THEN
+	TestFalse("Schema database not loaded", bSuccess);
```

Comment:
Is there no 'test cleanup' and 'test prepare' functionality in Unreal?

Reply:
I've added SchemaTestFixture that prepares the environment and cleans it up in constructor/destructor.